File: app/src/main/java/com/example/phonebook/repos/ContactsRepo.java
package com.example.phonebook.repos;

import com.example.phonebook.othersToBeSort.Contact;

import java.util.ArrayList;

public interface ContactsRepo {
    void addContact(Contact _newContact);
    void deleteContact(int _contactPosition);
    Contact getContact(int _contactPosition);
    ArrayList<Contact> getAllContacts();
    void updateContact(int _contactPosition, String _newName, String _newPone, String _newEmail, String _newAddress);
}

File: app/src/main/java/com/example/phonebook/data/repos/ContactsRepo.kt
package com.example.phonebook.data.repos

import androidx.lifecycle.LiveData
import com.example.phonebook.data.entities.Contact

import java.util.ArrayList

interface ContactsRepo {
    fun getLiveContactList(currentUser: String): LiveData<ArrayList<Contact>>
    fun getContactList(currentUser: String): ArrayList<Contact>
    fun saveContactList(user: String,  contactList: ArrayList<Contact>)
}
